debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -0,0 +1,392 @@
+ */
+public abstract class ValueTypeHelper {
+	protected static final Logger logger = Logger.getLogger(LoggerNames.LOGGER_INTERACTIVE_CONTEXT);
+	private static final long J9AccValueType = J9ConstantHelper.getLong(J9JavaAccessFlags.class, "J9AccValueType", 0);
+	private static final long J9ClassIsValueType = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassIsValueType", 0);
+	private static final long J9ClassLargestAlignmentConstraintDouble = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintDouble", 0);
+	private static final long J9ClassLargestAlignmentConstraintReference = J9ConstantHelper.getLong(J9JavaClassFlags.class, "J9ClassLargestAlignmentConstraintReference", 0);
+	private MethodHandle getFlattenedClassCachePointer = null;
+	private MethodHandle flattenedClassCache_offset = null;
+	private MethodHandle flattenedClassCache_NaS = null;
+	private MethodHandle flattenedClassCache_clazz = null;
+	private MethodHandle flattenedClassCache_cast = null;
+	private Class<?> flattenedClassCachePointer = null;
+	private static ValueTypeHelper helper = null;
+	
+	private static class ValueTypeSupportEnabledHelper extends ValueTypeHelper {
+		ValueTypeSupportEnabledHelper() {
+			super();
+		}
+	}
+	
+	private static class ValueTypeSupportDisabledHelper extends ValueTypeHelper {
+
+		ValueTypeSupportDisabledHelper() {
+			super();
+		}
+		
+		@Override
+		public J9ClassPointer findJ9ClassInFlattenedClassCacheWithFieldName(J9ClassPointer clazz, String fieldName) throws RuntimeException {
+			logger.log(Level.SEVERE, "Value types were not enabled in the JVM that produced this core file.");
+			throw new RuntimeException("Invalid operation");
+		}
+		
+		@Override
+		public J9ClassPointer[] findNestedClassHeirarchy(J9ClassPointer containerClazz, String[] nestingHierarchy) throws RuntimeException {
+			logger.log(Level.SEVERE, "Value types were not enabled in the JVM that produced this core file.");
+			throw new RuntimeException("Invalid operation");
+		}
+		
+		@Override
+		public J9ClassPointer findJ9ClassInFlattenedClassCacheWithSigName(J9ClassPointer clazz, String fieldSig) throws RuntimeException {
+			logger.log(Level.SEVERE, "Value types were not enabled in the JVM that produced this core file.");
+			throw new RuntimeException("Invalid operation");
+		}
+		
+		@Override
+		public boolean isFieldInClassFlattened(J9ClassPointer clazz, String fieldName) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean areValueTypesSupported() {
+			return false;
+		}
+		
+		@Override
+		public boolean isRomClassAValueType(J9ROMClassPointer romClass) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean isJ9ClassAValueType(J9ClassPointer clazz) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean isFlattenableFieldSignature(String signature) {
+			return false;
+		}
+	}
+	
+	
+	private ValueTypeHelper() {
```

Comment:
This constructor should not be `private`.
The initialization of these fields should not be done for `ValueTypeSupportDisabledHelper`.